parser/parser.go

```diff
@@ -25,7 +27,7 @@ type (
 var tokEOF = lexer.EOF
 
 var (
-	keywordParsers = map[token.Type]parserfn{}
+	keywordParsers map[token.Type]parserfn
```

Comment:
I think there will be some minor conflicts between my changes on the parser and yours...but not something too terrible to fix